### ISSUE TITLE
HDF5ExampleGen

### ISSUE THREAD

**Issue by CharlemagneBrain** [tensorflow/tfx#5811]
For models taking as inputs pre-processed files in hdf5 format, it would be nice to have a component that reads data from HDF5 files and converts them into TFX examples that can be used in our TFX pipeline, thus leading to the generation of interpretable statistics for data analysis, validation, anomaly detection.

**System information**

- TFX Version (you are using): 1.12.0
- Environment in which you plan to use the feature (e.g., Local
  (Linux/MacOS/Windows), Interactive Notebook, Google Cloud, etc..): Linux - Ubuntu 22

- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
ExampleGen is limited in the formats it supports

**Will this change the current API? How?**
Not specially, but it will extend application of TFX when data format is hdf5

**Who will benefit with this feature?** : Developpers, PhD Fellows, any person who want use TFX where data after preprocessing are in hdf5 format

**Do you have a workaround or are completely blocked by this?** : YES

**Name of your Organization (Optional)** : IRD

Thank you

**Comment by singhniraj08** [comment]
@CharlemagneBrain,

Currently, [TFX ExampleGen](https://www.tensorflow.org/tfx/guide/examplegen) supports external data sources like CSV, TFRecord, Avro, Parquet and BigQuery.

You can use [scripts](https://gist.github.com/shijianjian/93b52ae96137fcd3d63edd92eb3f5046) to convert HDF5 format to TFRecord format and feed data to your pipeline or create a [custom ExampleGen](https://www.tensorflow.org/tfx/guide/examplegen#custom_examplegen) for your pipeline to accept HDF5 format data by referring to [custom component tutorials](https://www.tensorflow.org/tfx/guide/custom_component).

Thank you!

**Comment by CharlemagneBrain** [comment]
Hello @singhniraj08 .
There is an error in the code at tf.python_io.TFRecordWriter(). 
I fixed it and here is the[link](https://gist.github.com/CharlemagneBrain/31ed6f5d3b1efbaf50c1ae6a4ce7e119),
Maybe you should then **merge** the code.

**Comment by singhniraj08** [comment]
@CharlemagneBrain,

Thank you so much for your contribution. This script was made by one of the community developers and we don't have access to merge code to that script. As an alternative, in future community can refer your code [hdf5_to_TFRecords.py](https://gist.github.com/CharlemagneBrain/31ed6f5d3b1efbaf50c1ae6a4ce7e119) for HDF5 ExampleGen implementation.

Thank you!

**Comment by CharlemagneBrain** [comment]
Yes, no worries. From this code, I was able to adapt it for ecg data of dimensions (N, 4096,12) whose labels are in a csv file.
So thanks to you.
